### DISCUSSION TITLE
Using Parcel as a dev server

### DISCUSSION THREAD

**Discussion by mscoobby** [cypress-io/cypress#16977] [Component Testing] [answered]
Hello everyone,
We've started using Cypress for testing our React based application and it works great but we are using Parcel to bundle the application and it would be nice if we don't have to introduce webpack to the project. I've looked around and tried different things but could not manage to make Cypress component testing work without webpack. I would appreciate any information on the topic.

**Comment by JessicaSachs** [comment, marked as answer]
Hey!

[We actually have support for Vite](https://docs.cypress.io/guides/component-testing/introduction#Experimental) (+ rollup) and I'd love to get support for Parcel.

Can you send me a repository with your normal development flow? I don't know how folks usually develop Parcel apps.

For component testing, we just need some sort of a dev server. Even something like a s[tatic express server](https://github.com/cypress-io/cypress-component-testing-examples/pull/26/files#diff-d8a2cb0e80218f3fa784fe5faf46577e18fec8ad6aa14023b3fce7d5e99f4bd8R14-R74) would work just fine.

Essentially, Cypress treats each spec as a standalone "page", so we need to have a harness that respects that contract and correctly returns a "compiled" file where your spec is essentially a `main.js` file.

In Webpack, that's particularly difficult because we need to support a lot of Webpack's proprietary bundling features... However, but for different bundlers, it's generally pretty easy to get it working.

Anyway, shoot me over an example repo and I'll take a look 👍🏻.

**Reply by JessicaSachs** [reply]
Here's an issue to track it. https://github.com/cypress-io/cypress/issues/17798

**Reply by maxMidius** [reply]
Hello -  Wondering if this issue is progressing.  I am also an  ex-webpack   - now a happy  Parcel user.
Thank you  -
